src/amo/components/CollectionManager/index.js

```diff
@@ -89,6 +93,14 @@ export class CollectionManagerBase extends React.Component<Props, State> {
       this.setState({
         addonAddedStatus: props.hasAddonBeenAdded ?
           ADDON_ADDED_STATUS_SUCCESS : null,
+      }, () => {
+        if (props.hasAddonBeenAdded) {
+          setTimeout(() => {
```

Comment:
I think we could inject `setTimeout` as a default prop and pass a function that executes directly in the test cases.

Reply:
hey @willdurand  - do you mean something like this?

```javascript
  static defaultProps = {
    setTimeout: () => {},
  };
```

I guess I am not totally clear of the benefit of this..? let me know if i am missing something or if something like this is ok:


```javascript:
componentWillReceiveProps(props: Props) {
    ....
 
    if (hasAddonBeenAdded !== hasAddonBeenAddedNew) {
      this.setState({
        addonAddedStatus: props.hasAddonBeenAdded ?
          ADDON_ADDED_STATUS_SUCCESS : null,
      });
    }
    
    // If the message status has been set, we'll reset the message status after 5 seconds
    if (hasAddonBeenAddedNew && hasAddonBeenAddedNew !== hasAddonBeenAdded) {
      setTimeout(this.resetMessageStatus, MESSAGE_RESET_TIME);
    }
  }
```

Reply:
I mean having a prop called `setTimeout` that is configured with `setTimeout` :D

In the tests, we can pass `(func) => func()` to that prop.